open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3348,211 @@ components:
           type: integer
           format: int64
 
+    BooleanTypeValue:
+      type: boolean
+      example: true
+
+    IntegerTypeValue:
+      type: integer
+      example: 42
+
+    LongTypeValue:
+      type: integer
+      format: int64
+      example: 9223372036854775807
+
+    FloatTypeValue:
+      type: number
+      format: float
+      example: 3.14
+
+    DoubleTypeValue:
+      type: number
+      format: double
+      example: 123.456
+
+    DecimalTypeValue:
+      type: string
+      description:
+        "Decimal types are serialized as strings using BigDecimal to maintain precision. For scales greater than 
+        or equal to zero, `toPlainString` ensures a non-exponential representation. Negative scales utilize the 
+       `toString`, resulting in exponential notation"
+      example:
+        positiveScale: "123.4500"
+        zeroScale: "2"
+        negativeScale: "2E+20"
+
+    StringTypeValue:
+      type: string
+      example: "hello"
+
+    UUIDTypeValue:
+      type: string
+      description:
+        "UUID types are serialized as a string"
```

Comment:
is this always a string in UUID4 format? can we also add restrictions about things like string length, etc. through OpenAPI spec?

Reply:
I was a little hesitant about providing the `format: uuid` to this property, since other UUID fields in the rest spec didn't apply this. The code base points to using UUID4 everywhere through the utils in java. When applying thie format property any codegen against this yaml will use the respective UUID class. For example, `java.util.UUID` for java and the `UUID` type for python which can infer UUID4. Was there any reason why this property isn't applied on other uuid fields?